Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -251,6 +263,11 @@ def network_create_engine_wrapper(self, pstate, initialdlstatus=None):
             atp["duplicate_is_error"] = True
             atp["hops"] = self.get_hops()
 
+            if share_mode:
+                atp["flags"] = lt.add_torrent_params_flags_t.flag_share_mode
+
+            self.set_checkpoint_disabled(checkpoint_disabled)
```

Comment:
It's been like that since I forked this branch from Elric's branch. I think it because we want to set that as soon as possible since `checkpoint()` can be called anytime. However, `network_create_engine_wrapper` is called from `restart`, so we need to re-set this as well. CMIIW